harmony_service_lib/provenance.py

```diff
@@ -0,0 +1,35 @@
+"""Utility functionality for generating provenance metadata."""
+from __future__ import annotations
+
+from datetime import datetime
+
+try:
+    # datetime.UTC added in Python 3.11
+    from datetime import UTC
+except ImportError:
+    from datetime import timezone
+    UTC = timezone.utc
+
+
+def get_updated_history_metadata(
+    service_name: str,
+    service_version: str,
+    existing_history: str | None = None,
+) -> str:
+    """Create updated the history global attribute.
+
+    This function primarily ensures the correct formatting of the history
+    string, and is agnostic to the format of the input file.
+
```

Comment:
It looks like most of the docstrings in the library document the parameters (though I don't like that they duplicate the type information already in the signature). It's probably worth adding descriptions to each parameter for this function to help service authors.

Reply:
Okay - I've added those descriptions in [4ebb910](https://github.com/nasa/harmony-service-lib-py/pull/60/commits/4ebb910d88410a07b6832d9b083807e61f301d62).